aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,477 @@
+---
+title: What's new in ASP.NET Core 3.0
+author: rick-anderson
+description: Learn about the new features in ASP.NET Core 3.0.
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/17/2019
+uid: aspnetcore-3.0
+---
+# What's new in ASP.NET Core 3.0
+
+This article highlights the most significant changes in ASP.NET Core 3.0 with links to relevant documentation.
+
+## ASP.NET Core 3.0 only runs on .NET Core 3.0
+
+As of ASP.NET Core 3.0, .NET Framework is no longer a supported target framework. Projects targeting .NET Framework can continue in a fully supported fashion using the [.NET Core 2.1 LTS release](https://www.microsoft.com/net/download/dotnet-core/2.1). ASP.NET Core 2.1 is supported until August 21, 2021. For more information, see [this GitHub announcement](https://github.com/aspnet/Announcements/issues/324).
+
+See [Port your code from .NET Framework to .NET Core](/dotnet/core/porting/) for migration information.
+
+## Use the ASP.NET Core shared framework
+
+The [ASP.NET Core 3.0 shared framework](https://natemcmaster.com/blog/2018/08/29/netcore-primitives-2/), contained in the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app), no longer requires an explicit `<PackageReference />` element in the project file. The shared framework is automatically referenced when using the `Microsoft.NET.Sdk.Web` SDK in your project file:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+```
+
+## Assemblies removed from the ASP.NET Core shared framework
+
+The most notable assemblies removed from the ASP.NET Core 3.0 shared framework are:
+
+* [Newtonsoft.Json](https://www.nuget.org/packages/Newtonsoft.Json/) (Json.NET). To add Json.NET to ASP.NET Core 3.0, see [Add Newtonsoft.Json-based JSON format support](xref:web-api/advanced/formatting#add-newtonsoftjson-based-json-format-support). ASP.NET Core 3.0 introduces `System.Text.Json` for reading and writing JSON. For more information, see [New JSON serialization](#json) in this document.
+* [Entity Framework Core](/ef/core/)
+
+For a complete list of assemblies removed from the shared framework, see [Assemblies being removed from Microsoft.AspNetCore.App 3.0](https://github.com/aspnet/AspNetCore/issues/3755). For more information on the motivation for this change, see [Breaking changes to Microsoft.AspNetCore.App in 3.0](https://github.com/aspnet/Announcements/issues/325) and [A first look at changes coming in ASP.NET Core 3.0](https://devblogs.microsoft.com/aspnet/a-first-look-at-changes-coming-in-asp-net-core-3-0/).
+
+<a name="json"></a>
+
+### New JSON serialization
+
+ASP.NET Core 3.0 includes `System.Text.Json`:
+
+* Reads and writes JSON asynchronously.
+* Is optimized for UTF-8 text.
+* Typically higher performance than `Newtonsoft.Json`.
+
+## gRPC
+
+See <xref:grpc/index>.
```

Comment:
Yup, that's a gRPC link all right.

Do you want any other feedback from me? 😋

Reply:
More seriously I think a one paragraph description of gRPC is and what we're shipping, like Razor components has, would be useful.

Reply:
Can you draft that? You can leave it as a comment and I'll wordsmith and commit.

Reply:
[gRPC](https://grpc.io/) is a popular, high-performance RPC (remote procedure call) framework that offers an opinionated contract-first approach to API development. It uses modern technologies such as HTTP/2 for transport, and Protocol Buffers as the interface description language and binary serialization format. gRPC provides features such as authentication, bidirectional streaming and flow control, and cancellation and timeouts.

gRPC functionality in ASP.NET Core 3.0 includes:

* [Grpc.AspNetCore](https://www.nuget.org/packages/Grpc.AspNetCore) &ndash; An ASP.NET Core framework for hosting gRPC services. gRPC on ASP.NET Core integrates with standard ASP.NET Core features like logging, dependency injection (DI), authentication and authorization.
* [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client) &ndash; A gRPC client for .NET Core that builds upon the familiar `HttpClient`.
* [Grpc.Net.ClientFactory](https://www.nuget.org/packages/Grpc.Net.ClientFactory) &ndash; gRPC client integration with `HttpClientFactory`.